lib/tree.rb

```diff
@@ -61,7 +86,7 @@ def tree_data
       diameter_sum += apple.diameter
     end
 
-    avg_diameter = # It's up to you to calculate the average diameter for this harvest.
+    avg_diameter = basket.size > 0 ? diameter_sum*1.0/basket.size : 0
```

Comment:
Remember that it is impossible for there to not be any apples, bc just before this on line 71, we age the tree until it has apples. Bc of this, the basket will have apples in it, thus doesn't need to be checked against zero.